### PR TITLE
Handle 404/ Invalid page request errors

### PR DESCRIPTION
**Pull request by drshrey**

### CODE REVIEW

**Comment by drshrey** [comment]
screenshot 2018-03-20 13 31 28

**Comment by drshrey** [comment]
addresses https://github.com/codeamp/panel/issues/35

File: src/components/Project/index.js [hunk 1]
```diff
@@ -118,6 +119,10 @@ export default class Project extends React.Component {
     if(loading){
       return (<Loading />)
     }
+    // handle invalid project
+    if(!project){
```
Comment (amoskyler) [inline src/components/Project/index.js]: Shouldn't be return this 404 page any time some route isn't handled, not only when the project doesn't exist?

Comment (drshrey) [reply src/components/Project/index.js]: yup, that's what line 153 does

Comment (amoskyler) [reply src/components/Project/index.js]: I think I'm just missing the render flow of this. If we are rendering 404 on any unhandled route, why do we explicitly call it for the project?

Comment (drshrey) [reply src/components/Project/index.js]: because project is used later on before render for setting other properties

Comment (drshrey) [reply src/components/Project/index.js]: If we don't do this, we'll get an application error: "project undefined"

Comment (amoskyler) [reply src/components/Project/index.js]: Gotcha. Will this flash a view of the 404 if the project hasn't been pulled yet? Or does the `if loading(` clause take care of it?

There's probably a way of attaching route paths to resource ids, so when a resource ID is not found, the route automatically interprets it as a 404. Not sure how to implement it, but just something to watch out for.

Comment (drshrey) [reply src/components/Project/index.js]: ^ I agree, that's a good refactor solution. We could include as something to match in our route path and then show 404 as the default component if it isn't matched.

**Pull request merged**